url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/robot/SOCRobotClient.java

```diff
@@ -0,0 +1,1880 @@
+     * @param mes  the message
+     */
+    protected void handleSTATUSMESSAGE(SOCStatusMessage mes)
+    {
+        final int sv = mes.getStatusValue();
+        if ((sv != 0) || ! printedInitialWelcome)
+        {
+            System.err.println("Robot " + getNickname() + ": Status "
+                + sv + " from server: " + mes.getStatus());
+            if (sv == 0)
+                printedInitialWelcome = true;
+        }
+    }
+
+    /**
+     * handle the "join game authorization" message
+     * @param mes  the message
+     */
+    protected void handleJOINGAMEAUTH(SOCJoinGameAuth mes)
+    {
+        gamesPlayed++;
+
+	final String gaName = mes.getGame();
+
+	SOCGame ga = new SOCGame(gaName, true, (Hashtable) gameOptions.get(gaName));
+        games.put(gaName, ga);
+        CappedQueue brainQ = new CappedQueue();
+        brainQs.put(gaName, brainQ);
+
+        // TODO Added by Ryan
+
+        // Uses our AI instead of theirs
+        // SOCRobotBrain rb = new SOCRobotBrain(this, currentRobotParameters, ga, brainQ);
+       /* int strategy= currentRobotParameters.getStrategyType();
+        SOCRobotBrain rb;
+        if (strategy== SOCRobotDM.OUR_STRATEGY){
+        	rb = new SOCOurBrain(this, currentRobotParameters, ga, brainQ);
+        } else {
+        	rb= new SOCRobotBrain(this, currentRobotParameters, ga, brainQ);
+        } */
+       // SOCOurBrain rb = new SOCOurBrain(this, currentRobotParameters, ga, brainQ);
+        
+
+        // Uses our AI instead of theirs, or vice versa
+        SOCRobotBrain rb;
+        if(useTheirRobot)
+        	rb = new SOCRobotBrain(this, currentRobotParameters, ga, brainQ);
+        else
+        	rb = new SOCOurBrain(this, currentRobotParameters, ga, brainQ);
+        robotBrains.put(gaName, rb);
+    }
+
+    /**
+     * handle the "join game" message
+     * @param mes  the message
+     */
+    protected void handleJOINGAME(SOCJoinGame mes) {}
+
+    /**
+     * handle the "game members" message, which indicates the entire game state has now been sent.
+     * If we have a {@link #seatRequests} for this game, sit down now.
+     * @param mes  the message
+     */
+    protected void handleGAMEMEMBERS(SOCGameMembers mes)
+    {
+        /**
+         * sit down to play
+         */
+        Integer pn = (Integer) seatRequests.get(mes.getGame());
+
+        try
+        {
+            //wait(Math.round(Math.random()*1000));
+        }
+        catch (Exception e)
+        {
+            ;
+        }
+
+        if (pn != null)
+        {
+            put(SOCSitDown.toCmd(mes.getGame(), nickname, pn.intValue(), true));
+        } else {
+        	System.out.println(this.nickname);
+            System.err.println("** Cannot sit down: Assert failed: null pn for game " + mes.getGame());
+        }
+    }
+
+    /**
+     * handle the "game text message" message
+     * @param mes  the message
+     */
+    protected void handleGAMETEXTMSG(SOCGameTextMsg mes)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 59. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1105-1117 into a separate method.